js/reference.js

```diff
@@ -109,16 +129,24 @@ var ERMrest = (function(module) {
      * reference to server-side resources could change.
      *
      * Usage:
-     *  Clients _do not_ directly access this constructor. 
+     *  Clients _do not_ directly access this constructor.
      *  See {@link ERMrest.resolve}.
      * @memberof ERMrest
      * @class
      * @param {!string} uri The `URI` for this reference.
      */
-    function Reference(uri) {
+    function Reference(uri, location) {
         this._uri = uri;
+        var context = module._parse(uri, location);
```

Comment:
the parser should be called directly in the `resolve` method above. this should not be pushed down into the `Reference` constructor